cmd/local.go

```diff
@@ -385,10 +385,9 @@ func local(cmd *cobra.Command, args []string) {
 		time.Sleep(5 * time.Millisecond)
 	}
 
-	if len(lo.Models) > 0 {
-		if viper.GetBool("saveConfig") {
-			configlib.SaveConfig(lo.Models[0].Nonmem.OriginalPath)
-		}
+	//If set to save config and the config value is not populated
+	if viper.GetBool("saveConfig") && len(viper.GetString("config")) == 0 {
```

Comment:
The `config` value is never serialized down to the `l.Nonmem.Configuration` struct on purpose. Conceptually this completely avoids potential loops for configurations being loaded. As such we reach directly to viper to see what the value is. If the value is not am empty string, even if `saveConfig` we won't save the file.